making_change/making_change.py

```diff
@@ -4,6 +4,16 @@
 
 def making_change(amount, denominations):
   pass 
+# USD - $1, $5, $10, etc.
+currency = [1, 5, 10, 20, 50, 100]
+if amount == 0:
+    return 1
+if amount == 0:
+    Return 0
+if len(denominations) == 0 and amount > 0:
+    return 0
+else:
+    return making_change(amount - denominations[-1], denominations) + making_change(amount - denominations[:1])
 
```

Comment:
This was a really tough one. Did you watch the solution lecture? I think it would be hard to do this without nested loops somehow.